x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/get_filter.ts

```diff
@@ -50,6 +54,7 @@ interface GetFilterArgs {
   savedId: string | undefined | null;
   services: AlertServices;
   index: string[] | undefined | null;
+  lists?: RuleAlertParams['lists'];
```

Comment:
question mark isn't needed. This should be:

```ts
lists: RuleAlertParams['lists'];
```

You will hit that tests need some `undefined` values passed into them like this example:

```ts
    test('returns a saved query if given a type of query', async () => {
      const filter = await getFilter({
        type: 'saved_query',
        filters: undefined,
        language: undefined,
        query: undefined,
        savedId: 'some-id',
        services: servicesMock,
        index: ['auditbeat-*'],
        lists: undefined,
      });
      expect(filter).toEqual({
        bool: {
          filter: [
            { bool: { minimum_should_match: 1, should: [{ match: { 'host.name': 'linux' } }] } },
          ],
          must: [],
          must_not: [],
          should: [],
        },
      });
    });
```

But that's a lot better to do as if people keep adding in question mark optionals to places it starts creating more and more bugs where some code paths are missed to include `undefined` (including test paths).

Optionals make it easy to make the TypeScript issues go away but introduce more runtime bugs and less tests for things. I just changed it locally and fixed the tests and it works well without the optional 👍

Reply:
Updated the code and tests to remove the use of the `?`. Per my other response - I'll be sure to try and avoid using it where not absolutely necessary.